fleet/redis/consumer/streams.go

```diff
@@ -147,6 +152,32 @@ func (es eventStore) handleDatasetRemove(ctx context.Context, e removeDatasetEve
 	return es.commsService.NotifyGroupDatasetRemoval(ag, e.datasetID, e.policyID)
 }
 
+func decodeDatasetUpdate(event map[string]interface{}) updateDatasetEvent {
+	return updateDatasetEvent{
+		id:            read(event, "id", ""),
+		ownerID:       read(event, "owner_id", ""),
+		agentGroupID:  read(event, "group_id", ""),
+		datasetID:     read(event, "dataset_id", ""),
+		policyID:      read(event, "policy_id", ""),
+		valid:         readBool(event, "valid", false),
+		turnedValid:   readBool(event, "turned_valid", false),
+		turnedInvalid: readBool(event, "turned_invalid", false),
+	}
+}
+
+func (es eventStore) handleDatasetUpdate(ctx context.Context, e updateDatasetEvent) error {
+	ag, err := es.fleetService.ViewAgentGroupByIDInternal(ctx, e.agentGroupID, e.ownerID)
```

Comment:
this should be done in the block below so that we avoid a database lookup unless we really need it